packages/jaeger-ui/src/components/DeepDependencyGraph/Graph.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react';
+import { DirectedGraph, LayoutManager } from '@jaegertracing/plexus';
+import _map from 'lodash/map';
+import { History as RouterHistory, Location } from 'history';
+import queryString from 'query-string';
+
+import GraphModel from '../../model/ddg/Graph';
+import { encode } from '../../model/ddg/visibility-codec';
+
+import { PathElem, TDdgModel } from '../../model/ddg/types';
+
+type TProps = {
+  history: RouterHistory;
+  location: Location;
+  ddgModel: TDdgModel;
+  visKey?: string;
+};
+
+export default class Graph extends Component<TProps> {
+  private graphModel: GraphModel;
+  private layoutManager: LayoutManager;
+
+  constructor(props: TProps) {
+    super(props);
+    const { ddgModel } = props;
+    const { distanceToPathElems } = ddgModel;
+    this.graphModel = new GraphModel({ ddgModel });
```

Comment:
Why not pass the `ddgModel` in as a param instead of a property on an options object?

Reply:
this makes it easier to add more arguments in the future, such as `visibilityHeuristic`